modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2532,11 +2554,12 @@ def binary_op(self, op, right_frame, join_type="outer"):
                 left_parts, op, right_parts[0]
             )
         )
-
+        if not make_column_reindex:
+            joined_columns = self.columns.join(right_frame.columns, how=join_type)
```

Comment:
could you please explain why?

Reply:
I rewrote this piece of code. `joined_columns` is now obtained from `_copartition` as before.